library/ZendService/Twitter/Media.php

```diff
@@ -0,0 +1,251 @@
+<?php
+/**
+ * @see       https://github.com/zendframework/ZendService_Twitter for the canonical source repository
+ * @copyright Copyright (c) 2017 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license   https://github.com/zendframework/ZendService_Twitter/blob/master/LICENSE.md New BSD License
+ */
+
+namespace ZendService\Twitter;
+
+use Zend\Http\Client as Client;
+use ZendService\Twitter\Response as Response;
+
+/**
+ * Twitter Media Uploader base class
+ *
+ * @author Cal Evans <cal@calevans.com>
+ */
+class Media
+{
+    const UPLOAD_BASE = 'https://upload.twitter.com';
+
+    const UPLOAD_PATH = '/1.1/media/upload.json';
+
+    /**
+     * Array of basic data to be stored by this class.
+     *
+     * @var Array
+     */
+    protected $data = [];
+
+    /**
+     * The number of bytes to send to Twitter at one time.
+     * @var Integer
+     */
+    protected $chunkSize = (1024 * 1024) * 4;
+
+    /**
+     * Constructor
+     *
+     * @param  null|string $imageFilename
+     * @param  string $consumer
+     */
+    public function __construct($imageFilename = null, $media_type = '')
```

Comment:
For new features, we should *REALLY* use PHP 7.1 as a base for development. That means proper parameter hints. Alternatively, assertions about types need to be added to the code.